### PR TITLE
*: tiny optimize by avoid sprintf

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: ref #56649

Problem Summary: tiny optimize by avoid sprintf

### What changed and how does it work?

before:
image

This PR:

image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
lgtm

File: pkg/types/datum.go [hunk 1]
```diff
@@ -564,19 +564,23 @@ func (d *Datum) TruncatedStringify() string {
 		if len(str) > maxLen {
 			// This efficiently returns the truncated string without
 			// less possible allocations.
-			return fmt.Sprintf("%s...(len:%d)", str[:maxLen], len(str))
+			return str[:maxLen] + "...(len:" + strconv.Itoa(len(str)) + ")"
```
Comment (ngaut) [inline pkg/types/datum.go:567]: Use string builder to concat string.

File: pkg/types/datum.go [hunk 2]
```diff
@@ -564,19 +564,23 @@ func (d *Datum) TruncatedStringify() string {
...
 		}
 		return str
 	case KindMysqlJSON:
 		// For now we can only stringify then truncate.
 		str := d.GetMysqlJSON().String()
 		if len(str) > maxLen {
-			return fmt.Sprintf("%s...(len:%d)", str[:maxLen], len(str))
+			return str[:maxLen] + "...(len:" + strconv.Itoa(len(str)) + ")"
```
Comment (ngaut) [inline pkg/types/datum.go:574]: Ditto.

File: pkg/types/datum.go (hunk 1 above)
Comment (crazycs520) [reply pkg/types/datum.go]: done.

**Pull request merged**